### PR TITLE
AP_Motors: Tradheli-incorporate DDFP reverse capability

### PR DESCRIPTION
**Pull request by bnsgeyer**
It was discovered that, when employing a Direct Drive Fixed Pitch tail on a helicopter with a rotor that spins counter clockwise (when viewed from above), the reversing the servo channel that controls the DDFP motor causes the motor to go to max speed when arming before motor interlock enabled.  This occurs because the pwm is high due to the servo output being reversed.  However the output must be reversed for the proper operation of the tail because increase in motor speed provides left yaw.  Obviously this is unsafe for the motor to go to max speed when armed before motor interlock.

Other options were explored before resorting to code to fix this issue.  A code change was the only option.  It comes down to the output of the yaw control is -1 to 1 where positive values of yaw control correspond to yaw right.  For servo output this corresponds to 1000 pwm for full left control and 2000 for full right control.  Reversing the servo to change control will never work because the servo immediately outputs 2000 pwm once armed and regardless of the direction of rotation of the motor and the orientation of the propeller, the ESC will always command full speed when the heli is armed.

Therefore a reverse bit had to be added for DDFP tails.  This has been functionally tested in the SITL.

### CODE REVIEW

File: libraries/AP_Motors/AP_MotorsHeli_Single.cpp [hunk 1]
```diff
@@ -135,6 +135,13 @@ const AP_Param::GroupInfo AP_MotorsHeli_Single::var_info[] = {
     // @Increment: 1
     AP_SUBGROUPINFO(_swashplate, "SW_", 20, AP_MotorsHeli_Single, AP_MotorsHeli_Swash),
 
+    // @Param: DDFP_REVERSED
```
Comment (tridge) [inline libraries/AP_Motors/AP_MotorsHeli_Single.cpp]: DDFP_YAW_DIR instead?
with values Clockwise, CounterClockwise

**Comment by tridge** [comment]
Rob suggested using tail type value instead

**Comment by bnsgeyer** [comment]
@rmackay9 I rebased this on master and made the suggested changes from the Dev Call. I functionally checked the changes and it all works well. Please take one last look over it before I merge. Also I don’t understand why some of the checks failed. Any help with that would be appreciated. Thanks!

**Comment by rmackay9** [comment]
LGTM, merged!  Hope it's OK I merged it so I can include in Copter-4.0.0-rc3.

**Comment by rmackay9** [comment]
this has been added to Copter-4.0.0-rc3

**Pull request merged**